controllers/machineset_controller.go

```diff
@@ -386,9 +386,10 @@ func shouldExcludeMachine(machineSet *clusterv1.MachineSet, machine *clusterv1.M
 
 // adoptOrphan sets the MachineSet as a controller OwnerReference to the Machine.
 func (r *MachineSetReconciler) adoptOrphan(machineSet *clusterv1.MachineSet, machine *clusterv1.Machine) error {
+	patch := client.MergeFrom(machine)
```

Comment:
deepcopy?

Reply:
Not needed in this case, since we only update the Metadata of Machines in the MachineSet controller.